event-based-simulators/main/oeeAPI.py

```diff
@@ -152,4 +153,39 @@ def get_simulator_external_ids(self):
         if ids:
             return self.c8y_api.get_external_ids(ids)
         log.warning(f'Didnt find any simulators: {ids}')
-        return []    
\ No newline at end of file
+        return []  
+    
+    def add_or_update_shiftplan(self, shiftplan):
+        locationId = shiftplan["locationId"]
+        for timeslot in shiftplan["recurringTimeslots"]:
+            if not self.add_timeslot(locationId, timeslot):
+                self.update_timeslot(locationId, timeslot, timeslot["id"])
```

Comment:
I think we agreed to not update existing timeslots as the user might intentionally update a shiftplan...

Reply:
Done